pre_commit_hooks/mixed_line_ending.py

```diff
@@ -0,0 +1,250 @@
+
+    def __init__(self, opt_name, line_ending_enum):
+        self.opt_name = opt_name
+        self.line_ending_enum = line_ending_enum
+
+
+class MixedLineDetection(Enum):
+    NOT_MIXED = 1, False, None
+    UNKNOWN = 2, False, None
+    MIXED_MOSTLY_CRLF = 3, True, LineEnding.CRLF
+    MIXED_MOSTLY_LF = 4, True, LineEnding.LF
+    MIXED_MOSTLY_CR = 5, True, LineEnding.CR
+
+    def __init__(self, index, mle_found, line_ending_enum):
+        # TODO hack to prevent enum overriding
+        self.index = index
+        self.mle_found = mle_found
+        self.line_ending_enum = line_ending_enum
+
+
+VERBOSE_OPTION_TO_LOGGING_SEVERITY = {
+    0: logging.WARNING,
+    1: logging.INFO,
+    2: logging.DEBUG,
+}
+
+
+ANY_LINE_ENDING_PATTERN = re.compile(
+    # match either
+    b'(' +
+    # \r\n
+    LineEnding.CRLF.regex.pattern +
+    # or \n
+    b'|' + LineEnding.LF.regex.pattern +
+    # or \r
+    b'|' + LineEnding.CR.regex.pattern +
+    b')'
+)
+
+
+def mixed_line_ending(argv=None):
+    options = _parse_arguments(argv)
+
+    logging.basicConfig(format='%(levelname)s: %(message)s',
+                        level=options['logging_severity'])
+    logging.debug('mixed_line_ending: options = %s', options)
+
+    filenames = options['filenames']
+    fix_option = options['fix']
+
+    _check_filenames(filenames)
+
+    if fix_option == MixedLineEndingOption.NO:
+        return _process_no_fix(filenames)
+    elif fix_option == MixedLineEndingOption.AUTO:
+        return _process_fix_auto(filenames)
+    # when a line ending character is forced with --fix option
+    else:
+        return _process_fix_force(filenames, fix_option.line_ending_enum)
+
+
+def _parse_arguments(argv=None):
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        '-f',
+        '--fix',
+        choices=[m.opt_name for m in MixedLineEndingOption],
+        default=MixedLineEndingOption.AUTO.opt_name,
+        help='Replace line ending with the specified. Default is "auto"')
+    parser.add_argument('filenames', nargs='*', help='Filenames to fix')
+    parser.add_argument(
+        '-v',
+        '--verbose',
+        action="count",
+        default=0,
+        help='Increase output verbosity')
+    args = parser.parse_args(argv)
+
+    fix = None
+    if args.fix == MixedLineEndingOption.NO.opt_name:
+        fix = MixedLineEndingOption.NO
+    elif args.fix == MixedLineEndingOption.CRLF.opt_name:
+        fix = MixedLineEndingOption.CRLF
+    elif args.fix == MixedLineEndingOption.LF.opt_name:
+        fix = MixedLineEndingOption.LF
+    else:
+        fix = MixedLineEndingOption.AUTO
+
+    args.verbose = min(args.verbose, 2)
+    severity = VERBOSE_OPTION_TO_LOGGING_SEVERITY.get(args.verbose)
+
+    options = {'fix': fix, 'filenames': args.filenames,
+               'logging_severity': severity}
+
+    return options
+
+
+def _check_filenames(filenames):
+    logging.debug('_check_filenames: filenames = %s', filenames)
```

Comment:
Well, it helped me a lot to [understand why the tests didn't run](https://travis-ci.org/nagromc/pre-commit-hooks/jobs/251655622#L263). It's up to you.

Do you want me to remove the `-vv` verbose mode? If so, maybe we should replace `logging.info` by `print` since it will remain only a single logging level (`INFO`).